### PR TITLE
sim: support LCD emulation over X11; correctly decouple LCD/FB from X11/NX

### PR DESCRIPTION
**Pull request by v01d**
## Summary

This PR adds support for simulated LCD device emulated over an X11 window, same as simulated FB. Moreover, configuration for choosing graphics/input for sim is improved: options were better organized in submenu and now you can choose whether to simulated FB or LCD. This choice is also independent of enabling X11 emulation or not (if not, these are dummy drivers, as intended originally). Input support is indeed dependant on X11, but this also now allows for eventually providing simulated input not necessarily via X11. Finally, some of this options (and part of implementation) where unnecessarily tied to CONFIG_NX.

## Impact

Improve graphics simulation on sim. Config symbol names where kept the same to avoid breakage (only dependencies and menu organization change).

## Testing

Retested lvgldemo over framebuffer. LCD also tested using own application.
Ideally lvgldemo should be extended to use either LCD or FB, depending on what is available/chosen.

### CODE REVIEW

**Comment by v01d** [comment]
@xiaoxiang781216 After this PR I will extend LCD interface as discussed. This simulated LCD is useful to validate it without hardware.

File: arch/sim/Kconfig [hunk 1]
```diff
@@ -257,28 +245,54 @@ config SIM_FBBPP
...
+
+choice
+  prompt "Graphics Device"
+  default SIM_LCDDRIVER
+  ---help---
+    Choose which kind of graphics device to emulate
+
+config SIM_LCDDRIVER
+	bool "LCD device"
+	default y
```
Comment (xiaoxiang781216) [inline arch/sim/Kconfig:258]: don't need, line 252 is enough

File: arch/sim/Kconfig [hunk 2]
```diff
@@ -257,28 +245,54 @@ config SIM_FBBPP
...
+	depends on LCD
+	---help---
+		Emulate an LCD driver
+
+config SIM_FRAMEBUFFER
+	bool "Framebuffer"
+	default n
```
Comment (xiaoxiang781216) [inline arch/sim/Kconfig:265]: remove too

**Comment by btashton** [comment]
Ah this will be nice, I was just messing with driving an i2c display via the simulator.

File: arch/sim/Kconfig (hunk 1 above)
Comment (v01d) [reply arch/sim/Kconfig]: done

File: arch/sim/Kconfig (hunk 2 above)
Comment (v01d) [reply arch/sim/Kconfig]: done

**Comment by v01d** [comment]
I justed pushed with the recommended changes, refreshed configs (as now SIM_FRAMEBUFFER is a default) and made some other minor improvements

**Comment by v01d** [comment]
I've also fixed some other configs which assumed VIDEO_FB was on by default. Also, I found some bad math in the putrun() implementation for simulated LCD

**Comment by v01d** [comment]
Ok, CI seems happy now

File: arch/sim/src/Makefile [hunk 3]
```diff
@@ -125,10 +125,14 @@ ifeq ($(CONFIG_RTC_DRIVER),y)
   CSRCS += up_rtc.c
 endif
 
-ifeq ($(CONFIG_NX_LCDDRIVER),y)
+ifeq ($(CONFIG_SIM_LCDDRIVER),y)
   CSRCS += up_lcd.c
 else
-  CSRCS += up_framebuffer.c
+  ifeq ($(CONFIG_SIM_FRAMEBUFFER),y)
```
Comment (xiaoxiang781216) [inline arch/sim/src/Makefile:131]: change to "else ifeq ($(CONFIG_SIM_FRAMEBUFFER),y)"?

Comment (v01d) [reply arch/sim/src/Makefile:131]: if you don't have other observations can be leave it like this for now? I'm juggling various branches and I have to go back and rebase changes each time. I already have ready the putarea() PR waiting for this one.

Comment (xiaoxiang781216) [reply arch/sim/src/Makefile:131]: Sure.

**Pull request merged**